gtsam/slam/EssentialMatrixFactor.h

```diff
@@ -267,28 +269,128 @@ class EssentialMatrixFactor3: public EssentialMatrixFactor2 {
    * @param E essential matrix
    * @param d inverse depth d
    */
-  Vector evaluateError(const EssentialMatrix& E, const double& d,
-      boost::optional<Matrix&> DE = boost::none, boost::optional<Matrix&> Dd =
-          boost::none) const override {
+  Vector evaluateError(
+      const EssentialMatrix& E, const double& d,
+      boost::optional<Matrix&> DE = boost::none,
+      boost::optional<Matrix&> Dd = boost::none) const override {
     if (!DE) {
       // Convert E from body to camera frame
       EssentialMatrix cameraE = cRb_ * E;
       // Evaluate error
       return Base::evaluateError(cameraE, d, boost::none, Dd);
     } else {
       // Version with derivatives
-      Matrix D_e_cameraE, D_cameraE_E; // 2*5, 5*5
+      Matrix D_e_cameraE, D_cameraE_E;  // 2*5, 5*5
       EssentialMatrix cameraE = E.rotate(cRb_, D_cameraE_E);
       Vector e = Base::evaluateError(cameraE, d, D_e_cameraE, Dd);
-      *DE = D_e_cameraE * D_cameraE_E; // (2*5) * (5*5)
+      *DE = D_e_cameraE * D_cameraE_E;  // (2*5) * (5*5)
       return e;
     }
   }
 
-public:
+ public:
   GTSAM_MAKE_ALIGNED_OPERATOR_NEW
 };
 // EssentialMatrixFactor3
 
-}// gtsam
+/**
+ * Binary factor that optimizes for E and calibration K using the algebraic
+ * epipolar error (K^-1 pA)'E (K^-1 pB). The calibration is shared between two
+ * images.
+ *
+ * Note: As correspondences between 2d coordinates can only recover 7 DoF,
+ * this factor should always be used with a prior factor on calibration.
+ * Even with a prior, we can only optimize 2 DoF in the calibration. So the
+ * prior should have a noise model with very low sigma in the remaining
```

Comment:
Thanks for the PR, guys. should we give a definition for what range constitutes a "very low" sigma?

Reply:
I think this is relative, it depends on the sigma that is not low. There are examples in the unit tests, which are still empirical. Not sure if there is an accurate answer that can be briefly documented.